pylops/basicoperators/directionalderivative.py

```diff
@@ -71,25 +59,40 @@ def FirstDirectionalDerivative(
     operator with :math:`\mathbf{v}` along the main diagonal.
 
     """
-    Gop = Gradient(dims, sampling=sampling, edge=edge, kind=kind, dtype=dtype)
-    if v.ndim == 1:
-        Dop = Diagonal(v, dims=[len(dims)] + list(dims), axis=0, dtype=dtype)
-    else:
-        Dop = Diagonal(v.ravel(), dtype=dtype)
-    Sop = Sum(dims=[len(dims)] + list(dims), axis=0, dtype=dtype)
-    ddop = LinearOperator(Sop * Dop * Gop)
-    ddop.dims = ddop.dimsd = dims
-    ddop.sampling = sampling
-    return ddop
-
-
-def SecondDirectionalDerivative(
-    dims: InputDimsLike,
-    v: NDArray,
-    sampling: int = 1,
-    edge: bool = False,
-    dtype: DTypeLike = "float64",
-) -> LinearOperator:
+
+    def __init__(self, dims: InputDimsLike,
+                 v: NDArray,
+                 sampling: int = 1,
+                 edge: bool = False,
+                 kind: str = "centered",
+                 dtype: DTypeLike = "float64",
+                 name: str = 'F'):
+        self.sampling = sampling
+        self.edge = edge
+        self.kind = kind
+        self.v = v
+
+        super().__init__(Op=self._calc_first_ddop(dims=dims, sampling=sampling, edge=edge, kind=kind, dtype=dtype, v=v),
+                         dims=dims, dtype=dtype, name=name)
```

Comment:
No need to supply dims and dtype here, they will be inherited from self._calc_first_ddop. Could you change to this for readability? (in the others as well)

```python
Op = self._calc_first_ddop(dims=dims, sampling=sampling, edge=edge, kind=kind, dtype=dtype, v=v)
super().__init__(Op=Op, name=name)
```